From: Yosry Ahmed <yosryahmed@google.com>
Subject: mm-zswap-add-zswap_never_enabled-fix
Date: Tue, 11 Jun 2024 23:37:52 +0000

Expose zswap_never_enabled() in the header for the swapin work to use
it later.

Link: https://lkml.kernel.org/r/Zmjf0Dr8s9xSW41X@google.com
Signed-off-by: Yosry Ahmed <yosryahmed@google.com>
Cc: Barry Song <baohua@kernel.org>
Cc: Chengming Zhou <chengming.zhou@linux.dev>
Cc: Chris Li <chrisl@kernel.org>
Cc: David Hildenbrand <david@redhat.com>
Cc: Johannes Weiner <hannes@cmpxchg.org>
Cc: Matthew Wilcox (Oracle) <willy@infradead.org>
Cc: Nhat Pham <nphamcs@gmail.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 include/linux/zswap.h |    6 ++++++
 mm/zswap.c            |    2 +-
 2 files changed, 7 insertions(+), 1 deletion(-)

--- a/include/linux/zswap.h~mm-zswap-add-zswap_never_enabled-fix
+++ a/include/linux/zswap.h
@@ -36,6 +36,7 @@ void zswap_memcg_offline_cleanup(struct
 void zswap_lruvec_state_init(struct lruvec *lruvec);
 void zswap_folio_swapin(struct folio *folio);
 bool zswap_is_enabled(void);
+bool zswap_never_enabled(void);
 #else
 
 struct zswap_lruvec_state {};
@@ -64,6 +65,11 @@ static inline bool zswap_is_enabled(void
 {
 	return false;
 }
+
+static inline bool zswap_never_enabled(void)
+{
+	return false;
+}
 
 #endif
 
--- a/mm/zswap.c~mm-zswap-add-zswap_never_enabled-fix
+++ a/mm/zswap.c
@@ -137,7 +137,7 @@ bool zswap_is_enabled(void)
 	return zswap_enabled;
 }
 
-static bool zswap_never_enabled(void)
+bool zswap_never_enabled(void)
 {
 	return !static_branch_maybe(CONFIG_ZSWAP_DEFAULT_ON, &zswap_ever_enabled);
 }
_
